Assets/MixedRealityToolkit.Services/InputAnimation/Editor/InputRecordingWindow.cs

```diff
@@ -0,0 +1,351 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit;
+using Microsoft.MixedReality.Toolkit.Utilities.Editor;
+using UnityEngine;
+using UnityEngine.Playables;
+using UnityEngine.SceneManagement;
+using UnityEditor;
+using System;
+using System.IO;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// Tools for recording and playing back input animation in the Unity editor.
+    /// </summary>
+    public class InputRecordingWindow : EditorWindow
+    {
+        private InputAnimation animation
+        {
+            get { return PlaybackService?.Animation; }
+            set { if (PlaybackService != null) PlaybackService.Animation = value; }
+        }
+
+        private string loadedFilePath = "";
+
+        private IMixedRealityInputRecordingService recordingService = null;
+        private IMixedRealityInputRecordingService RecordingService =>
+            recordingService ??
+            (recordingService = MixedRealityToolkit.Instance.GetService<IMixedRealityInputRecordingService>());
```

Comment:
@davidkline-ms is this the preferred way to access a service?

ixedRealityToolkit.Instance.GetService<IMixedRealityInputRecordingService>()

 

If not, what is the guidance for the right way to acquire services? I always look for code to copy/paste but would prefer a page I can refer to with "here are a bunch of exmaples of the preferred way to acquire services". For example, I often point people to https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Contributing/CodingGuidelines.html for instructions on the preferred way to have properties with public getters/setters.

Reply:
Ah yes, i should update these to MixedRealityServiceRegistry as well. Things have changed a lot since i first wrote this code.

Reply:
fixed.